changelog.d/6487.misc

```diff
@@ -0,0 +1 @@
+Pass in `Database` object to data stores.
```

Comment:
this isn't a great changelog entry, but we probably want to change all the datastore refactoring PRs to group them together, which is most easily done at release time.

Reply:
Agreed, though I find it difficult to write sensible change logs for internal refactors tbh